fix: fix hover information for File Description variable

amanPras opened the pull request:
## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Test hover information at line 12 and line 15
```cobol
       IDENTIFICATION DIVISION.
       PROGRAM-ID. TESTHVR.
       ENVIRONMENT DIVISION.
       INPUT-OUTPUT SECTION.
       FILE-CONTROL.
            SELECT ACCT-FILE ASSIGN TO ACCTFIL
                ORGANIZATION INDEXED
                ACCESS IS DYNAMIC
                RECORD KEY IS ACCT-KEY.
       DATA DIVISION.
       FILE SECTION.
       FD  ACCT-FILE
           LABEL RECORDS ARE STANDARD.
       01  ACCT-RECORD.
         02  ACCT-KEY              PIC X(5).
         02  FILLER                PIC X(378).
``` 
Below hover code for FD node at line 12
```
       DESCRIPTION 
       ----------- 
       FD  ACCT-FILE
       LABEL RECORDS ARE STANDARD.
 
       FILE-CONTROL
       ------------ 
       ASSIGN TO ACCTFIL
       ORGANIZATION INDEXED
       ACCESS IS DYNAMIC
       RECORD KEY IS ACCT-KEY 
``` 
Below hover code at line 15
```
       
         01 ACCT-RECORD.
           02 ACCT-KEY PIC X(5).
``` 

## Checklist:
- [x] Each of my commits contains one meaningful change
- [x] I have performed rebase of my branch on top of the development
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

amanPras committed:
fix: fix hover information for File Description variable

Signed-off-by: amanprashant <aman.prashant@broadcom.com>

amanPras commented on server/engine/src/main/java/org/eclipse/lsp/cobol/service/delegates/hover/DefinedAndUsedStructureHoverProvider.java:
```diff
@@ -131,14 +135,19 @@ private static Stream<Hover> getHoverStream(
     StringBuilder trimmedHoverContent = new StringBuilder();
+    String prefix = StringUtils.repeat(" ", startCharIndex);
     for (int i = 0; i < limit; i++) {
+      if (!lines[i].startsWith(prefix)) {
```
is this needed or should we simply add the prefix?

slavek-kucera replied:
I think it should be added unconditionally (just like it is removed unconditionally).

amanPras replied:
updated

amanPras committed:
fix: fix hover information for File Description variable

Signed-off-by: amanprashant <aman.prashant@broadcom.com>

amanPras committed:
test: fix tests

Signed-off-by: amanprashant <aman.prashant@broadcom.com>

amanPras committed:
test: fix integration tests

Signed-off-by: amanprashant <aman.prashant@broadcom.com>

slavek-kucera commented:
Two details:
- There seems to be an extra empty line at the top of the hover text.
- Should the `DESCRIPTION`, `FILE-CONTROL` and the separators be either cobol comments or outside of the ` ```cobol ``` ` language blocks?

amanPras committed:
fix: take hover information from the source

Signed-off-by: amanprashant <aman.prashant@broadcom.com>

amanPras committed:
Revert "fix: take hover information from the source"

This reverts commit cb089f2172d47ccc0c02ca005a8101b078397fef.

amanPras committed:
fix: fix hover information for FD

changes:
1. fix hover information for FD
2. Keep cobol formating for hover text

Signed-off-by: amanprashant <aman.prashant@broadcom.com>

amanPras committed:
chore: format code

Signed-off-by: amanprashant <aman.prashant@broadcom.com>

slavek-kucera commented on server/common/src/main/java/org/eclipse/lsp/cobol/common/model/tree/variable/VariableNode.java:
```diff
@@ -139,17 +144,27 @@ public void extendLocality(Position newEndPosition) {
     for (String parentLine : parentsDescription()) {
-      lines.add(prepend(prefix.toString(), parentLine));
-      prefix.append(PREFIX);
+      lines.add(new MarkupContent(MarkupKind.MARKDOWN, prepend(prefix.toString(), parentLine)));
```
`prefix` is always empty at this point?

amanPras replied:
refactored to
```java
lines.add(new MarkupContent(MarkupKind.MARKDOWN, parentLine));
```
